File: 1796-second-largest-digit-in-a-string/1796-second-largest-digit-in-a-string.java
class Solution {
    public int secondHighest(String s) {
        int m=Integer.MIN_VALUE;
        int sm=-1;
        for(char c:s.toCharArray()){
            if(Character.isDigit(c)){
                int v=c-'0';
                if(m<v){
                    sm=m;
                    m=v;
                }
                else if(v<m && v>sm) sm=v;
            }
        }
        if(sm==Integer.MIN_VALUE) return -1;
        return sm;
    }
}

File: 1796-second-largest-digit-in-a-string/1796-second-largest-digit-in-a-string.cpp
class Solution {
public:
    int secondHighest(string s) {
        int m=INT_MIN;
        int sm=-1;
        for(int i=0;i<s.size();i++){
            if(s[i]-'0'>=0 && s[i]-'0'<10){
                int v=s[i]-'0';
                if(m<v){
                    sm=m;
                    m=v;
                }
                else if(v<m && v>sm) sm=v;
            }
        }
        if(sm==INT_MIN) return -1;
        return sm;
    }
};